Source/WebCore/dom/TrustedType.cpp

```diff
@@ -302,14 +303,35 @@ ExceptionOr<String> requireTrustedTypesForPreNavigationCheckPasses(ScriptExecuti
         : nullString());
 }
 
-ExceptionOr<bool> canCompile(ScriptExecutionContext& scriptExecutionContext, JSC::CompilationType compilationType, String codeString, JSC::JSValue bodyArgument)
+ExceptionOr<bool> canCompile(ScriptExecutionContext& scriptExecutionContext, JSC::CompilationType compilationType, String codeString, const JSC::ArgList& args)
 {
-    VM& vm = scriptExecutionContext.vm();
-
-    if (bodyArgument.isObject())
-        return JSTrustedScript::toWrapped(vm, bodyArgument) ? true : false;
+    if (compilationType == CompilationType::Function) {
+        VM& vm = scriptExecutionContext.vm();
+        auto scope = DECLARE_THROW_SCOPE(vm);
+
+        bool isTrusted = true;
+
+        for (size_t i = 0; i < args.size(); i++) {
+            auto arg = args.at(i);
+            if (!arg.isObject()) {
+                isTrusted = false;
+                break;
+            }
+            if (auto trustedScript = JSTrustedScript::toWrapped(vm, arg)) {
+                if (!trustedScript || trustedScript->toString() != arg.toWTFString(scriptExecutionContext.globalObject())) {
+                    RETURN_IF_EXCEPTION(scope, Exception { ExceptionCode::ExistingExceptionError });
```

Comment:
is is not a correct place to emit exception check. When the above code pass with == then no exception check exist.
Every time we run `toWTFString` etc., just after that, exception check should be done.

Reply:
I've added another one after the if statement is that correct?